### ISSUE TITLE
Accessibility: preventOpenOnFocus doesn't work in mobile or ipad

### ISSUE THREAD

**Issue by arunkumar509** [Hacker0x01/react-datepicker#2452]
**Describe the bug**
preventOpenOnFocus just works when we do a keyboard focus. It doesn't work in mobile or ipad. As soon as user taps on input field date picker opens.

**To Reproduce**

1. Load datepicker with preventOpenOnFocus props set to true [in mobile/tab]
2. Tap on the datepicker
3. See error

**Expected behavior**
Expected date picker not to show up. But it does.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

Device: [e.g. > iPhone6 ]
Browser [e.g.  safari]

**Additional context**
We are working on implmenting date picker in our website. While validating for Accessibility the date picker and the virtual calendar are confusing users. So the ask is to make accessible users type the date instead of selecting from the date picker.
